src/sail_c_backend/sail_plugin_c.ml

```diff
@@ -83,6 +83,19 @@ let c_options =
       Arg.String (fun prefix -> C_backend.opt_prefix := prefix),
       "prefix generated C functions"
     );
+    (Flag.create ~prefix:["c"] "cpp", Arg.Set C_backend.opt_cpp, "generate C++ struct instead of global C functions");
```

Comment:
`-c --c-cpp` seems a bit counterintuitive. We could set this up so it's just `--cpp` by creating a new target, and the `--c-cpp-struct-name` would just be `--cpp-struct-name`.

Reply:
Probably easiest way would be to do something like:
```
let c_target is_cpp out_file { ast; effect_info; env; default_sail_dir; _ } =
```
and then change the last two lines of `sail_plugin_c` into:
```
   Target.register ~name:"c" ~options:c_options ~rewrites:c_rewrites ~supports_abstract_types:true
    ~supports_runtime_config:true (c_target false);
  Target.register ~name:"cpp" ~options:cpp_options ~rewrites:c_rewrites ~supports_abstract_types:true
    ~supports_runtime_config:true (c_target true)
```
The `cpp_options` list would just contain the new options for the C++ generation.